include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,330 @@
+#include "kodlab_mjbots_sdk/log.h"
+
+namespace kodlab {
+
+/**
+ * @brief Manager for `Behavior` objects
+ * @details The behavior manager class is a container for storing numerous
+ *          behaviors for a given robot interface. It manages their selection
+ *          and transitions between them.
+ * @note By default, this class initializes its internal behaviors list with a
+ *       `kodlab::Offbehavior`. This is done for safety purposes so that, when
+ *       turned on, the robot does not command some undefined torques. This
+ *       default can be overwritten with the `SetDefaultBehavior` member.
+ * @tparam Robot[optional] derived `kodlab::RobotBase` class
+ */
+template<class Robot = kodlab::RobotBase>
+class BehaviorManager {
+  static_assert(std::is_base_of<kodlab::RobotBase, Robot>::value,
+                "Robot must have base kodlab::RobotBase.");
+
+ private:
+  /**
+   * @brief Robot that behaviors run on
+   */
+  std::shared_ptr<Robot> robot_;
+
+  /**
+   * @brief Behavior command that kills robot, default -1
+   */
+  const int KILL_ROBOT_IDX = -1;
+
+  /**
+   * @brief Name of the kill behavior
+   */
+  const std::string KILL_BEHAVIOR_NAME = "KILL";
+
+  /**
+   * @brief Behavior command that turns robot off, always 0
+   */
+  const int DEFAULT_IDX = 0;
+
+  /**
+   * @brief Name of the default behavior
+   */
+  std::string default_name_;
+
+  /**
+   * @brief Behaviors available to the robot
+   * @note The first element of this vector is always `OFF_BEHAVIOR`.  Added
+   *       behaviors are appended to the end.
+   */
+  std::vector<std::unique_ptr<Behavior<Robot>>> behaviors_;
+
+  /**
+   * @brief Map from behavior name to index in `behaviors_`
+   */
+  std::unordered_map<std::string, int> name_to_idx_map_;
+
+  /**
+   * @brief Names of behaviors in `behaviors_`
+   */
+  std::vector<std::string> names_;
+
+  /**
+   * @brief Index in `behaviors_` of the selected behavior
+   */
+  unsigned int selected_idx = 0;
+
+  /**
+   * @brief Index in `behaviors_` of the incoming behavior while transitioning
+   *        behaviors
+   */
+  int next_idx_ = 0;
+
+  /**
+   * @brief Indicates whether a behavior switch is in progress
+   */
+  bool switching_behaviors_ = false;
+
+  /**
+   * @brief Constructs and initializes a behavior
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   * @return unique pointer to the behavior if initialization was successful,
+   *         nullptr otherwise
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  std::unique_ptr<BehaviorType> ConstructBehavior(ConstructorArgs &&... args) {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = std::make_unique<BehaviorType>(args...);
+    if (b->Init()) {
```

Comment:
In the current paradigm, is init ever called not right after the constructor?

Reply:
No, currently `Init` is only called right after the constructor.  `Init` called outside the constructor as the [style guide](https://drake.mit.edu/styleguide/cppguide.html#Doing_Work_in_Constructors) discourages calling virtual methods in constructors (for good reason).  This separation is also maintained in case a `BehaviorManager` child wishes to call `Init` for all contained behaviors at the same time.